### ISSUE TITLE
[pkgconf] Build error complain ERROR: Could not detect Ninja v1.8.2 or newer

### ISSUE THREAD

**Issue by ywang2017** [microsoft/vcpkg#25202]
- OS:  arm64v8/ubuntu:latest container
- Compiler: revision
```
root@cc29c852d18e:/home/ec2-user# ninja --version
1.11.0.git

root@cc29c852d18e:/home/ec2-user#./vcpkg/vcpkg/vcpkg install pkgconf

Computing installation plan...
The following packages will be built and installed:
    pkgconf[core]:arm64-linux -> 1.8.0#2
Detecting compiler hash for triplet arm64-linux...
Restored 0 packages from /root/.cache/vcpkg/archives in 785.4 us. Use --debug to see more details.
Starting package 1/1: pkgconf:arm64-linux
Building package pkgconf[core]:arm64-linux...
-- Using community triplet arm64-linux. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: /home/ec2-user/vcpkg/vcpkg/triplets/community/arm64-linux.cmake
-- Using cached pkgconf-pkgconf-cef30268e1a3f79efd607c26abcf556aa314c9c4.tar.gz.
-- Cleaning sources at /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/src/6aa314c9c4-8bd74a463d.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/ec2-user/vcpkg/vcpkg/downloads/pkgconf-pkgconf-cef30268e1a3f79efd607c26abcf556aa314c9c4.tar.gz
-- Using source at /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/src/6aa314c9c4-8bd74a463d.clean
-- Getting CMake variables for arm64-linux-dbg
-- Getting CMake variables for arm64-linux-rel
-- Configuring arm64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:128 (message):
    Command failed: /usr/bin/python3 /home/ec2-user/vcpkg/vcpkg/installed/arm64-linux/tools/meson/meson.py -Dtests=false --buildtype 
plain --backend ninja --wrap-mode nodownload --native /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux.log --default-library static --libdir lib --native /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux-debug.log -Ddebug=true --prefix /home/ec2-user/vcpkg/vcpkg/packages/pkgconf_arm64-linux/debug --includedir ../include -Dcmake_prefix_path=['/home/ec2-user/vcpkg/vcpkg/installed/arm64-linux/debug','/home/ec2-user/vcpkg/vcpkg/installed/arm64-linux'] /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/src/6aa314c9c4-8bd74a463d.clean
    Working Directory: /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/arm64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/config-arm64-linux-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_meson.cmake:492 (vcpkg_execute_required_process)
  ports/pkgconf/portfile.cmake:9 (vcpkg_configure_meson)
  scripts/ports.cmake:145 (include)


Error: Building package pkgconf:arm64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `./vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+pkgconf
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[pkgconf]+Build+error    
including:
  package: pkgconf[core]:arm64-linux -> 1.8.0#2
    vcpkg-tool version: 2999-12-31-unknownhash
    vcpkg-scripts version: unknown

Additionally, attach any relevant sections from the log files above.
```
**Additional context**
 /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/config-arm64-linux-dbg-out.log
```
 more /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/config-arm64-linux-dbg-out.log     
WARNING: Recommend using either -Dbuildtype or -Doptimization + -Ddebug. Using both is redundant since they override each other. See: https://mesonbuild.com/Builtin-options.html#build-type-options
The Meson build system
Version: 0.60.2
Source dir: /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/src/6aa314c9c4-8bd74a463d.clean      
Build dir: /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/arm64-linux-dbg
Build type: native build
Project name: pkgconf
Project version: 1.8.0
C compiler for the host machine: /usr/bin/cc (gcc 11.2.0 "cc (Ubuntu 11.2.0-19ubuntu1) 11.2.0")
C linker for the host machine: /usr/bin/cc ld.bfd 2.38
Host machine cpu family: aarch64
Host machine cpu: aarch64
Checking for function "strlcat" : NO
Checking for function "strlcpy" : NO
Checking for function "strndup" : YES
Header <string.h> has symbol "strndup" : YES
Configuring config.h using configuration
Build targets in project: 2

pkgconf 1.8.0

  User defined options
    Native files     : /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux.log
                       /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux-debug.log
    backend          : ninja
    buildtype        : plain
    cmake_prefix_path: ['/home/ec2-user/vcpkg/vcpkg/installed/arm64-linux/debug','/home/ec2-user/vcpkg/vcpkg/installed/arm64-linux']
    debug            : true
    default_library  : static
    includedir       : ../include
    libdir           : lib
    prefix           : /home/ec2-user/vcpkg/vcpkg/packages/pkgconf_arm64-linux/debug
    wrap_mode        : nodownload
    tests            : false


ERROR: Could not detect Ninja v1.8.2 or newer

A full log can be found at /home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/arm64-linux-dbg/meson-logs/meson-log.txt       
root@cc29c852d18e:/home/ec2-user#
```

**Comment by JackBoosY** [comment]
Please also provide _/home/ec2-user/vcpkg/vcpkg/buildtrees/pkgconf/arm64-linux-dbg/meson-logs/meson-log.txt_.

**Comment by dipal004** [comment]
I have received the same erro. 
After installing the ninja using
$pip3 install ninja
Now it is solved.
You can have a try with this

**Comment by yuz1** [comment]
Build started at 2022-09-15T15:16:41.747885
Main binary: /usr/bin/python3
Build Options: -Dtests=false -Ddebug=false '-Dcmake_prefix_path=['"'"'/home/test/project/ABC/build/vcpkg_installed/arm64-linux'"'"','"'"'/home/test/project/ABC/build/vcpkg_installed/arm64-linux/debug'"'"']' -Dprefix=/home/test/project/vcpkg/packages/pkgconf_arm64-linux -Dlibdir=lib -Dbackend=ninja -Dbuildtype=plain -Ddefault_library=static -Dwrap_mode=nodownload '--native-file /home/test/project/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux.log' '--native-file /home/test/project/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux-release.log'
Python system: Linux
The Meson build system
Version: 0.62.1
Source dir: /home/test/project/vcpkg/buildtrees/pkgconf/src/6aa314c9c4-8bd74a463d.clean
Build dir: /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel
Build type: native build
Project name: pkgconf
Project version: 1.8.0
Sanity testing C compiler: /usr/bin/cc
Is cross compiler: False.
Sanity check compiler command line: /usr/bin/cc sanitycheckc.c -o sanitycheckc.exe -fPIC -O3 -DNDEBUG -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include -D_FILE_OFFSET_BITS=64 -L/home/test/project/ABC/build/vcpkg_installed/arm64-linux/lib
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/sanitycheckc.exe
C compiler for the host machine: /usr/bin/cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-10kylin2) 9.3.0")
C linker for the host machine: /usr/bin/cc ld.bfd 2.34
Sanity testing C compiler: /usr/bin/cc
Is cross compiler: False.
Sanity check compiler command line: /usr/bin/cc sanitycheckc.c -o sanitycheckc.exe -D_FILE_OFFSET_BITS=64
Sanity check compile stdout:

-----
Sanity check compile stderr:

-----
Running test binary command: /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/sanitycheckc.exe
C compiler for the build machine: /usr/bin/cc (gcc 9.3.0 "cc (Ubuntu 9.3.0-10kylin2) 9.3.0")
C linker for the build machine: /usr/bin/cc ld.bfd 2.34
Build machine cpu family: aarch64
Build machine cpu: aarch64
Host machine cpu family: aarch64
Host machine cpu: aarch64
Target machine cpu family: aarch64
Target machine cpu: aarch64
Running compile:
Working directory:  /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpt4z54jj0
Command line:  /usr/bin/cc -L/home/test/project/ABC/build/vcpkg_installed/arm64-linux/lib -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpt4z54jj0/testfile.c -o /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpt4z54jj0/output.exe -fPIC -O3 -DNDEBUG -D_FILE_OFFSET_BITS=64 -O0 

Code:
 #include <string.h>
#include <limits.h>

        #if defined __stub_strlcat || defined __stub___strlcat
        fail fail fail this function is not going to work
        #endif
        
int main(void) {
            void *a = (void*) &strlcat;
            long long b = (long long) a;
            return (int) b;
        }
Compiler stdout:
 
Compiler stderr:
 /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpt4z54jj0/testfile.c: In function 'main':
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpt4z54jj0/testfile.c:9:32: error: 'strlcat' undeclared (first use in this function); did you mean 'strncat'?
    9 |             void *a = (void*) &strlcat;
      |                                ^~~~~~~
      |                                strncat
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpt4z54jj0/testfile.c:9:32: note: each undeclared identifier is reported only once for each function it appears in

Running compile:
Working directory:  /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2
Command line:  /usr/bin/cc -L/home/test/project/ABC/build/vcpkg_installed/arm64-linux/lib -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2/testfile.c -o /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2/output.exe -fPIC -O3 -DNDEBUG -D_FILE_OFFSET_BITS=64 -O0 

Code:
 #include <string.h>
        int main(void) {

        /* With some toolchains (MSYS2/mingw for example) the compiler
         * provides various builtins which are not really implemented and
         * fall back to the stdlib where they aren't provided and fail at
         * build/link time. In case the user provides a header, including
         * the header didn't lead to the function being defined, and the
         * function we are checking isn't a builtin itself we assume the
         * builtin is not functional and we just error out. */
        #if !0 && !defined(strlcat) && !0
            #error "No definition for __builtin_strlcat found in the prefix"
        #endif

        #ifdef __has_builtin
            #if !__has_builtin(__builtin_strlcat)
                #error "__builtin_strlcat not found"
            #endif
        #elif ! defined(strlcat)
            __builtin_strlcat;
        #endif
        return 0;
        }
Compiler stdout:
 
Compiler stderr:
 /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2/testfile.c: In function 'main':
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2/testfile.c:12:14: error: #error "No definition for __builtin_strlcat found in the prefix"
   12 |             #error "No definition for __builtin_strlcat found in the prefix"
      |              ^~~~~
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2/testfile.c:20:13: error: '__builtin_strlcat' undeclared (first use in this function); did you mean '__builtin_strcat'?
   20 |             __builtin_strlcat;
      |             ^~~~~~~~~~~~~~~~~
      |             __builtin_strcat
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpki21y9g2/testfile.c:20:13: note: each undeclared identifier is reported only once for each function it appears in

Checking for function "strlcat" : NO 
Running compile:
Working directory:  /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmprds8mt3y
Command line:  /usr/bin/cc -L/home/test/project/ABC/build/vcpkg_installed/arm64-linux/lib -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmprds8mt3y/testfile.c -o /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmprds8mt3y/output.exe -fPIC -O3 -DNDEBUG -D_FILE_OFFSET_BITS=64 -O0 

Code:
 #include <string.h>
#include <limits.h>

        #if defined __stub_strlcpy || defined __stub___strlcpy
        fail fail fail this function is not going to work
        #endif
        
int main(void) {
            void *a = (void*) &strlcpy;
            long long b = (long long) a;
            return (int) b;
        }
Compiler stdout:
 
Compiler stderr:
 /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmprds8mt3y/testfile.c: In function 'main':
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmprds8mt3y/testfile.c:9:32: error: 'strlcpy' undeclared (first use in this function); did you mean 'strncpy'?
    9 |             void *a = (void*) &strlcpy;
      |                                ^~~~~~~
      |                                strncpy
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmprds8mt3y/testfile.c:9:32: note: each undeclared identifier is reported only once for each function it appears in

Running compile:
Working directory:  /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy
Command line:  /usr/bin/cc -L/home/test/project/ABC/build/vcpkg_installed/arm64-linux/lib -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy/testfile.c -o /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy/output.exe -fPIC -O3 -DNDEBUG -D_FILE_OFFSET_BITS=64 -O0 

Code:
 #include <string.h>
        int main(void) {

        /* With some toolchains (MSYS2/mingw for example) the compiler
         * provides various builtins which are not really implemented and
         * fall back to the stdlib where they aren't provided and fail at
         * build/link time. In case the user provides a header, including
         * the header didn't lead to the function being defined, and the
         * function we are checking isn't a builtin itself we assume the
         * builtin is not functional and we just error out. */
        #if !0 && !defined(strlcpy) && !0
            #error "No definition for __builtin_strlcpy found in the prefix"
        #endif

        #ifdef __has_builtin
            #if !__has_builtin(__builtin_strlcpy)
                #error "__builtin_strlcpy not found"
            #endif
        #elif ! defined(strlcpy)
            __builtin_strlcpy;
        #endif
        return 0;
        }
Compiler stdout:
 
Compiler stderr:
 /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy/testfile.c: In function 'main':
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy/testfile.c:12:14: error: #error "No definition for __builtin_strlcpy found in the prefix"
   12 |             #error "No definition for __builtin_strlcpy found in the prefix"
      |              ^~~~~
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy/testfile.c:20:13: error: '__builtin_strlcpy' undeclared (first use in this function); did you mean '__builtin_strcpy'?
   20 |             __builtin_strlcpy;
      |             ^~~~~~~~~~~~~~~~~
      |             __builtin_strcpy
/home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpxulnugjy/testfile.c:20:13: note: each undeclared identifier is reported only once for each function it appears in

Checking for function "strlcpy" : NO 
Running compile:
Working directory:  /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmp9ew9b2k7
Command line:  /usr/bin/cc -L/home/test/project/ABC/build/vcpkg_installed/arm64-linux/lib -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmp9ew9b2k7/testfile.c -o /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmp9ew9b2k7/output.exe -fPIC -O3 -DNDEBUG -D_FILE_OFFSET_BITS=64 -O0 

Code:
 #include <string.h>
#include <limits.h>

        #if defined __stub_strndup || defined __stub___strndup
        fail fail fail this function is not going to work
        #endif
        
int main(void) {
            void *a = (void*) &strndup;
            long long b = (long long) a;
            return (int) b;
        }
Compiler stdout:
 
Compiler stderr:
 
Checking for function "strndup" : YES 
Running compile:
Working directory:  /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpdcqoc96d
Command line:  /usr/bin/cc -I/home/test/project/ABC/build/vcpkg_installed/arm64-linux/include /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpdcqoc96d/testfile.c -o /home/test/project/vcpkg/buildtrees/pkgconf/arm64-linux-rel/meson-private/tmpdcqoc96d/output.obj -c -fPIC -O3 -DNDEBUG -D_FILE_OFFSET_BITS=64 -O0 

Code:
 
        #include <string.h>
        int main(void) {
            /* If it's not defined as a macro, try to use as a symbol */
            #ifndef strndup
                strndup;
            #endif
            return 0;
        }
Compiler stdout:
 
Compiler stderr:
 
Header <string.h> has symbol "strndup" : YES 
Configuring config.h using configuration
Build targets in project: 2
NOTICE: Future-deprecated features used:
 * 0.56.0: {'meson.build_root', 'meson.source_root'}

pkgconf 1.8.0

  User defined options
    Native files     : /home/test/project/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux.log
                       /home/test/project/vcpkg/buildtrees/pkgconf/meson-native-arm64-linux-release.log
    backend          : ninja
    buildtype        : plain
    cmake_prefix_path: ['/home/test/project/ABC/build/vcpkg_installed/arm64-linux','/home/test/project/ABC/build/vcpkg_installed/arm64-linux/debug']
    debug            : false
    default_library  : static
    libdir           : lib
    prefix           : /home/test/project/vcpkg/packages/pkgconf_arm64-linux
    wrap_mode        : nodownload
    tests            : false


ERROR: Unhandled python exception

    This is a Meson bug and should be reported!

**Comment by LilyWangLL** [comment]
This issue hasn’t been updated in 3 months, if it is still an issue, please reopen this issue.
